spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/PartitionCapableBinderTests.java

```diff
@@ -198,13 +200,14 @@ public void testPartitionedModuleSpEL() throws Exception {
 		catch (UnsupportedOperationException ignored) {
 		}
 
-		Message<Integer> message2 = MessageBuilder.withPayload(2)
+		Message<String> message2 = MessageBuilder.withPayload("2")
```

Comment:
It can, true, but on the receiving I still need to check for a String, I just thought it would be a bit more consistent.
In fairness I'd just send a byte if I was writing the test today, this was just to make sure it goes green with the new handling of payloads.